include/zephyr/drivers/adc/adc_voltage_divider.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_ADC_VOLTAGE_DIVIDER_H_
+#define ZEPHYR_INCLUDE_DRIVERS_ADC_VOLTAGE_DIVIDER_H_
+
+#include <zephyr/drivers/adc.h>
+
+struct voltage_divider_dt_spec {
+	const struct adc_dt_spec port;
+	uint32_t full_ohms;
+	uint32_t output_ohms;
+};
+
+/**
+ * @brief Get voltage divider information from devicetree.
+ *
+ * This returns a static initializer for a @p voltage_divider_dt_spec structure
+ * given a devicetree node.
+ *
+ * @param node_id Devicetree node identifier.
+ *
+ * @return Static initializer for an voltage_divider_dt_spec structure.
+ */
+#define VOLTAGE_DIVIDER_DT_SPEC_GET(node_id)                                                       \
+	{                                                                                          \
+		.port = ADC_DT_SPEC_GET(node_id),                                                  \
+		.full_ohms = DT_PROP_OR(node_id, full_ohms, 0),                                    \
```

Comment:
shouldn't `full-ohms` be required? full_ohms is >= output_ohms

Reply:
From what I recall `full-ohms` was spec'd as optional so that the binding can be used to define a divider where the higher resistor is variable, like a PTC/NTC or something. I think we should leave it like this right now and in a followup PR handle that functionality, and maybe even extend it so that the fixed one can be the upper leg. Or we could require both and have a separate binding for that case, what do you think?

Reply:
The current `voltage-divider` documentation has `full-ohms` as optional to allow for `voltage-divider` to be used as a resistance based sensor instead:
https://docs.zephyrproject.org/latest/build/dts/api/bindings/adc/voltage-divider.html

I suppose we could change the binding to have a separate binding specifically for resistance based sensors, though I don't know if that's going to break any existing projects.